docs/COMPONENT_INTEGRATION.md

```diff
@@ -0,0 +1,311 @@
+
+`NewControllerReconciler` utilizes a generic builder pattern, that supports defining various types of relationships and functionality:
+- resource ownership - using `.Owns()`
+- watching a resource - using `.Watches()`
+- reconciler actions - using `.WithAction()`
+	- this includes pre-implemented actions used commonly across components (e.g. manifest rendering), as well as custom, component-specific actions
+	- more details on actions are provided [below](#actions)
+
+The example pseudo-implementation could look as the following:
+```go
+func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ReconcilerFor(mgr, &componentApi.ExampleComponent{}).
+		Owns(...).
+		// ... add other necessary resource ownerships
+		Watches(...).
+		// ... add other necessary resource watches
+		WithAction(...).
+		// ... add custom actions if needed
+		// ... add mandatory common actions (e.g. manifest rendering, deployment, garbage collection)
+		Build(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+```
+
+##### Actions
+
+Actions are functions that define pieces of component reconciliation logic. Any action is expected to conform to the following signature:
+
+```go
+func exampleAction(ctx context.Context, rr *odhtypes.ReconciliationRequest) error
+```
+
+Such actions can be then introduced to the reconciler builder using `.WithAction()` calls.
+As seen in the existing component reconciler implementations, it would be recommended to include the action implementations in a separate file within the module, such as `<example_component_name>_controller_actions.go`.
+
+"Generic"/commonly-implemented actions for each of the currently integrated components include:
+- `initialize()` - to register paths to the component manifests
+- `devFlags()` - to override the component manifest paths according to the Dev Flags configuration 
+
+In addition, proper generic actions, intended to be used across the components, are provided as part of the operator implementation (located in `pkg/controller/actions`).
+These support:
+- (if necessary) creating pod security role binding
+- manifest rendering
+    - can additionally utilize caching
+- manifest deployment
+    - can additionally utilize caching
+- status updating
+- garbage collection
+	- **additional requirement - garbage collection action must always be called as the last action before the final `.Build()` call**
+
+In case the new component requires additional custom logic, custom actions can also be added to the builder via the respective `.WithAction()` calls.
+
+For practical examples of all the abovementioned functionality, please refer to the implementations within `controllers/components` directory.
+
+#### Register new component to the manager (DSC controller)
+
+Register the new component API to the manager as the following pseudo-implementation:
+```go
```

Comment:
```suggestion
```diff
```

Using a `diff` code block rather than a `go` code block like this will allow highlighting the line to be added, at the expense of language-specific syntax highlighting. I think it's a worthwhile tradeoff in this case (there might be other cases like that here where it might be applicable too).